libr/bin/format/mach0/coresymbolication.h

```diff
@@ -0,0 +1,85 @@
+#include <r_bin.h>
+#include <r_types.h>
+
+#ifndef _INCLUDE_R_BIN_CORESYMBOLICATION_H
+#define _INCLUDE_R_BIN_CORESYMBOLICATION_H
+
+typedef struct r_coresym_cache_element_hdr_t {
+	ut32 version;
+	ut32 size;
+	ut32 n_segments;
+	ut32 n_sections;
+	ut32 n_symbols;
+	ut32 n_lined_symbols;
+	ut32 n_line_info;
+	ut32 f;
+	ut32 g;
+	ut32 h;
+	ut32 file_name_off;
+	ut32 version_off;
+	ut32 k;
+	ut8 uuid[16];
+	ut32 cputype;
+	ut32 cpusubtype;
+	ut32 o;
+	ut32 strings_off;
+	ut32 p;
+} RCoreSymCacheElementHdr;
+
+typedef struct r_coresym_cache_element_segment_t {
+	ut64 paddr;
+	ut64 vaddr;
+	ut64 size;
+	ut64 vsize;
+	char name[17];
+} RCoreSymCacheElementSegment;
+
+typedef struct r_coresym_cache_element_section_t {
+	ut64 paddr;
+	ut64 vaddr;
+	ut64 size;
+	char name[256];
+} RCoreSymCacheElementSection;
+
+typedef struct r_coresym_cache_element_flc_t {
+	char file[512];
+	ut32 line;
+	ut32 col;
+} RCoreSymCacheElementFLC;
+
+typedef struct r_coresym_cache_element_line_info_t {
+	ut32 paddr;
+	ut32 size;
+	RCoreSymCacheElementFLC flc;
+} RCoreSymCacheElementLineInfo;
+
+typedef struct r_coresym_cache_element_symbol_t {
+	ut32 paddr;
+	ut32 size;
+	ut32 unk1;
+	char name[512];
```

Comment:
is this 512 limit imposed by the standard? or just a lazy way to avoid having to hold a heap pointer?

Reply:
as long as 17, 256 and 512 are used in several char arrays in this file would be better to have a define on top to specify this size in a generic way

Reply:
lazy

Reply:
but in theory is segment name + section name so i think i can shrink it